File: source/com/barrybecker4/simulation/trading/model/tradingstrategy/Transaction.java
/*
 * Copyright by Barry G. Becker, 2015. Licensed under MIT License: http://www.opensource.org/license.MIT
 */

package com.barrybecker4.simulation.trading.model.tradingstrategy;

/**
 * Contains the purchase price and how much was bought.
 * Immutable
 * @author Barry Becker
 */
public class Transaction {

    double stockPrice;
    double amount;
    double numShares;

    Transaction(double price, double amount) {
        this.stockPrice = price;
        this.amount = amount;
        this.numShares = amount / price;
    }

    public double getStockPrice() {
        return stockPrice;
    }

    public double getAmount() {
        return amount;
    }

    public double getNumShares() {
        return numShares;
    }

}

File: scala-source/com/barrybecker4/simulation/trading/model/tradingstrategy/Transaction.scala
// Copyright by Barry G. Becker, 2016-2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.simulation.trading.model.tradingstrategy

/**
  * Contains the purchase price and how much was bought.
  * @author Barry Becker
  */
case class Transaction private[tradingstrategy](stockPrice: Double, amount: Double) {
  val numShares = amount / stockPrice
}
